### PR TITLE
window smoothing of k-mer counts within unitigs

### PR DESCRIPTION
**Pull request by karasikov**

### CODE REVIEW

**Review by hmusta — CHANGES_REQUESTED** [review]

File: metagraph/src/common/algorithms.hpp [hunk 1]
```diff
@@ -216,6 +216,63 @@ namespace utils {
...
+
+        if (window_size <= 1)
+            return;
+
+        size_t left = window_size / 2;
+        size_t right = window_size - left - 1;
+
+        assert(left >= right);
+
+        if (right + 1 >= v.size()) {
```
[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/src/common/algorithms.hpp:231]: for window_size/2 < v.size() < window_size there are still non-trivial averaging on the sides of the array (see picture in the comment below).

File: metagraph/src/common/algorithms.hpp [hunk 2]
```diff
@@ -216,6 +216,63 @@ namespace utils {
...
 
+    template <typename T>
+    void smooth_vector(size_t window_size, std::vector<T> *v_p) {
+        auto &v = *v_p;
+
+        if (window_size <= 1)
+            return;
+
+        size_t left = window_size / 2;
+        size_t right = window_size - left - 1;
```
[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/src/common/algorithms.hpp:227]: How would you suggest changing it? But not something too long.

File: metagraph/src/common/algorithms.hpp [hunk 3]
```diff
@@ -216,6 +216,63 @@ namespace utils {
         return count_hist.back().first;
     }
 
+    template <typename T>
+    void smooth_vector(size_t window_size, std::vector<T> *v_p) {
```
[the remark this answers is not in the document]
Comment (karasikov) [reply metagraph/src/common/algorithms.hpp:229]: added a comment

File: metagraph/src/common/algorithms.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (hmusta) [reply metagraph/src/common/algorithms.hpp:236]: maybe something like `left_back` and `right_begin`? but that's not that much better. let's just keep it like this...

**Review by hmusta — APPROVED** [review]

**Pull request merged**